plugins/gaming.py

```diff
@@ -116,19 +116,79 @@ def dice(text, notice):
         return "{} ({})".format(total, ", ".join(rolls))
 
 
+choice_split_re = re.compile(r'\s*(?:, ?or|,|or)\s*', re.I)  # regex for splitting choices for choose command
+
+
 @hook.command
 def choose(text, event):
     """<choice1>, [choice2], [choice3], etc. - randomly picks one of the given choices
     :type text: str
     """
-    choices = re.findall(r'([^,]+)', text.strip())
+    choices = choice_split_re.split(text.strip())
+
     if len(choices) == 1:
-        choices = choices[0].split(' or ')
-        if len(choices) == 1:
+        event.notice_doc()
+        return
+
+    return random.choice([choice.strip() for choice in choices])
+
+
+# array of digits/characters indexed by int value
+val_char_map = [chr(x) for x in
+                [c for r in (range(ord('0'), ord('9') + 1), range(ord('a'), ord('z') + 1)) for c in r]]
+
+
+def order_bounds(value_a, value_b):
+    """Return the input bounds correctly ordered for randint function"""
+    if value_a < value_b:
+        return value_a, value_b
+
+    return value_b, value_a
+
+
+@hook.command("chooser", "chooserange")
+def chooserange(text, event, notice):
+    """<a> to <b>|<a> <b> - picks a random integer or character from the given range"""
+    choices = re.split(r'\s*(?:to| +)\s*', text)
+
+    if len(choices) == 2:
+        try:
+
+            try:
+                # attempt to parse as integer range
+                bound_a = int(choices[0])
+                bound_b = int(choices[1])
+
+            except ValueError:
+                # values aren't integers
+                bound_a = None
+                bound_b = None
+
+            if bound_a and bound_b:
+                lower_bound, upper_bound = order_bounds(bound_a, bound_b)
+                return str(random.randint(lower_bound, upper_bound))
+
+            if choices[1].isalpha() or choices[2].isalpha():
+                # 0-9 + A-Z
+                # get the integer index of the digits/characters
+                lower_bound, upper_bound = order_bounds(int(choices[0], 36), int(choices[1], 36))
+
+                if lower_bound < 0 or upper_bound > 36:
+                    notice("When using letters, bounds must be within the range 0-9 + A-Z")
+                    return
+
+                choice = random.randint(lower_bound, upper_bound)
+
+                return str(val_char_map[choice]).upper()
+
             event.notice_doc()
             return
+        except ValueError:
+            notice("Invalid range given: {}".format(text))
+            return
```

Comment:
What exactly is this `try-except` handling?

Reply:
Thanks for pointing it out, as it seems it doesn't do anything useful anymore. I think it was an artifact from trying to catch bad int conversions.